src/java.desktop/share/classes/sun/awt/SunToolkit.java

```diff
@@ -445,6 +445,9 @@ public static void postEvent(AppContext appContext, AWTEvent event) {
         if (event == null) {
             throw new NullPointerException();
         }
+        if (appContext == null) {
+            return;
+        }
```

Comment:
I think we should check why the appcontext is null here, the event comes from the native peer, so when the event was created the peer was there, if the peer exists the target should have an appconext as well.

Maybe the problem is that we try to jump on EDT twice? First time from CTrayIcon.postEvent-> SunToolkit.executeOnEventHandlerThread and then one more time via SunToolkit.postEvent().

Can deletion of the executeOnEventHandlerThread solve the problem?

Reply:
Or maybe it is a lack of synchronization?

Reply:
So what is the root cause of null context here?

Reply:
I'm checking to see if removing executeOnEventHandlerThread solves the issue. In this test the source of the appcontext (trayIcon) is deleted via an actionPerformed listener, so probably the deletion of the trayIcon comes first then the mouseEvent is trying to be posted onto the trayIcon and throwing an error

Reply:
if the SunToolkit.executeOnEventHandlerThread is removed, then the test doesn't fail, but the icon doesn't disappear (test fails)

Reply:
then it should be checked what is going on there. The context should not just become null for the existing component.

Reply:
I think the context became null because the component doesn't exist anymore

Reply:
Somehow we have null passed in.
So something looked up appcontext and got null.
Likely because some other code removed the app context from a map or similar. 
First step is to find exactly where that happened, then we can start to reason about how we managed to still post an event.

Reply:
Note that it does not fail with NPE but instead this code triggee the bug:
```
        AppContext eventContext = targetToAppContext(event.getSource());
        if (eventContext != null && !eventContext.equals(appContext)) {
            throw new RuntimeException("Event posted on wrong app context : " + event);
        }
```
The source of the event is TrayIcon, and the appconteext for it is non-null as well.

This is the code where we post an event:
```
    @Override
    public void dispose() {
        dummyFrame.dispose();

        if (popup != null) {
            popup.removeNotify();
        }

        LWCToolkit.targetDisposedPeer(target, this);
        target = null;

        super.dispose();
    }

    private void postEvent(final AWTEvent event) {
        SunToolkit.executeOnEventHandlerThread(target, new Runnable() {
            public void run() {
                SunToolkit.postEvent(SunToolkit.targetToAppContext(target), event);
            }
        });
    }
```

Note that we also call SunToolkit.targetToAppContext(target), and the target should have a non-null appcontext, because later we successfully fetch it.


So the problem here is not in the null appconteext, but that we set target to null in the peer in dispose(), at the time we still processing the events.
 - So we should check how all this should be synchronized, to prevent such race
 - Or we should not set target to null, this is why the bug is not reproduced on other platforms.

Reply:
I think we should set the target once and use it forever.

Reply:
ok, i'll try removing the target = null